src/hooks/useForm.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { ChangeEvent, FormEvent, MouseEventHandler, useState } from 'react';
+
+interface objectType {
+  [key: string]: string;
+}
+
+interface useFormProps<T> {
+  initialValues: T;
+  onSubmit: () => Promise<void>;
+  validate: (values: T) => T | objectType;
+}
+
+const useForm = <T extends objectType>({ initialValues, onSubmit, validate }: useFormProps<T>) => {
+  const [values, setValues] = useState<T>(initialValues);
+  const [errors, setErrors] = useState<T | objectType>({});

```

Comment:
```suggestion
  const [errors, setErrors] = useState<T>({} as T);
```
두 방법 중 어떤 게 더 나을까요..🙄
첫번째는 `T extends objectType | objectType`이라 타입이 중복되는 느낌이고, 두번째는 `as T`로 타입 단언하는 게 뭔가 찜찜하군요🥺

Reply:
as T로 가겠습니다! 타입 지정하는게 미흡해서 어렵군여 😥